### PR TITLE
fix: [PROD-11486] fix minus and NaN in BasincInputNumber

### PR DESCRIPTION
**Pull request by ardvi**

### CODE REVIEW

File: src/inputs/BasicInputs/BasicNumberInput/BasicNumberInput.js [hunk 1]
```diff
@@ -17,37 +17,63 @@ export const BasicNumberInput = (props) => {
...
 
   const handleChangeEvent = useCallback(
     (event) => {
-      setTextInput(event.target.value);
-      const inputValueAsNumber = parseFloat(event.target.value);
-      if (inputValueAsNumber !== value) changeNumberField(inputValueAsNumber);
+      const textValue = event.target.value;
+      const numValue = parseFloat(textValue);
+      if (!isNaN(numValue)) {
+        changeNumberField(parseFloat(numValue));
```
Comment (csm-thu) [inline src/inputs/BasicInputs/BasicNumberInput/BasicNumberInput.js:51]: I think `parseFloat(numValue)` should be only `numValue`, we don't need to parse it again :slightly_smiling_face:

**Review by nborde-CSM — APPROVED** [review]
One remark but LGTM :+1:

**Review by esasova — APPROVED** [review]
LGTM

File: src/inputs/BasicInputs/BasicNumberInput/BasicNumberInput.js [hunk 2]
```diff
@@ -17,37 +17,63 @@ export const BasicNumberInput = (props) => {
   const { id, label, tooltipText, value, textFieldProps, inputProps, changeNumberField, isDirty, ...otherProps } =
     props;
 
-  const [textInput, setTextInput] = useState(value.toString());
+  const convToStringValue = useCallback((numValue) => {
+    if (numValue === undefined || numValue === null) {
```
[the remark this answers is not in the document]
Comment (ardvi) [reply src/inputs/BasicInputs/BasicNumberInput/BasicNumberInput.js]: done

**Review by csm-thu — APPROVED** [review]
LGTM :+1:

**Pull request merged**